### PR TITLE
Introduce TOSA pass to convert all i48 types to i64

### PR DESCRIPTION
**Pull request by NatashaKnk**

### CODE REVIEW

**Comment by jpienaar** [comment]
How much of this are cases introduced by issues in flow above here? E.g., I looked at Conversion/TosaToLinalg/TosaToLinalg.cpp and tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc where 48-bit constants are being either created or documented to be fine to not use 48 bit. Is the goal here for this to be short term or longer term?

**Comment by NatashaKnk** [comment]
Hmm, good question. Ideally if the wonky types are introduces earlier we would like to fix that, but as far as I understand there are various situations that cause it. I think @rsuderman has some more context on this. 
I would like to say that we want to get rid of them all, and if there are only a few iit might be a reasonable effort. Otherwise though, this seems like a good catch-all that aligns better with team prioritization. If you think otherwise, I'd be happy to work on a more thorough solution.

**Comment by jpienaar** [comment]
I'm fine with this if this works until then (I pinged one of the TOSA folks too)

**Comment by rsuderman** [comment]
> How much of this are cases introduced by issues in flow above here? E.g., I looked at Conversion/TosaToLinalg/TosaToLinalg.cpp and tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc where 48-bit constants are being either created or documented to be fine to not use 48 bit. Is the goal here for this to be short term or longer term?

It is difficult to say. For today it would be reasonable to just convert all `i48` types to `i64` in tosa-to-linalg, though its already quit a complex pass. Post conversion feels fine to me for now and I do not expect anything too complex enough to cause issues later on.

We can always revisit this in the future but it is unlikely anyone will feel opinionated on this in the long run.

File: compiler/src/iree/compiler/InputConversion/TOSA/Converti48Toi64.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  }
+};
+
+static bool isIllegalType(Type type) {
+  if (type.isInteger(48))
+    return true;
+  if (auto shapedType = dyn_cast<ShapedType>(type)) {
+    return isIllegalType(shapedType.getElementType());
+  }
+  return false;
```
Comment (rsuderman) [inline compiler/src/iree/compiler/InputConversion/TOSA/Converti48Toi64.cpp]: Nit: You can remove the `type.isInteger(48)` case above and replace `return false;` with `return type.isInteger(48);`

**Review by rsuderman — APPROVED** [review]

**Review by jpienaar — APPROVED** [review]

**Pull request merged**